src/widgets/scrollbar.rs

```diff
@@ -446,198 +449,177 @@ impl<'a> Scrollbar<'a> {
-        //                   v
-        //   ┌───────────────┐
-        //   │               ║<──────── track_start (y1)
-        //   │               █
-        //   │               █
-        //   │               ║
-        //   │               ║<──────── track_end   (y2)
-        //   └───────────────┘
-        //
-        // For ScrollbarOrientation::HorizontalBottom
-        //
-        //   ┌───────────────┐
-        //   │               │
-        //   │               │
-        //   │               │
-        //   └═══███═════════┘<──────── track_axis  (y)
-        //    ^             ^
-        //    │             └────────── track_end   (x2)
-        //    │
-        //    └──────────────────────── track_start (x1)
-        //
-
-        // Find track_start, track_end, and track_axis
-        let area = self.get_track_area(area);
-        let (track_start, track_end, track_axis) = self.get_track_start_end(area);
-
-        if self.should_not_render(track_start, track_end, state.content_length) {
+        if state.content_length == 0 {
             return;
         }
 
-        let (thumb_start, thumb_end) = self.get_thumb_start_end(state, (track_start, track_end));
-
-        for i in track_start..track_end {
-            let (style, symbol) = if i >= thumb_start && i < thumb_end {
-                (self.thumb_style, self.thumb_symbol)
-            } else if let Some(track_symbol) = self.track_symbol {
-                (self.track_style, track_symbol)
-            } else {
-                continue;
-            };
-
-            if self.is_vertical() {
-                buf.set_string(track_axis, i, symbol, style);
-            } else {
-                buf.set_string(i, track_axis, symbol, style);
-            }
-        }
-
-        if let Some(s) = self.begin_symbol {
-            if self.is_vertical() {
-                buf.set_string(track_axis, track_start - 1, s, self.begin_style);
-            } else {
-                buf.set_string(track_start - 1, track_axis, s, self.begin_style);
-            }
-        };
-        if let Some(s) = self.end_symbol {
-            if self.is_vertical() {
-                buf.set_string(track_axis, track_end, s, self.end_style);
-            } else {
-                buf.set_string(track_end, track_axis, s, self.end_style);
+        let track_axis = self.get_track_axis(area);
+
+        let (track_start_len, thumb_len, track_end_len) = self.get_track_lens(area, state);
+
+        let begin = (self.begin_symbol, self.begin_style);
+        let track = (self.track_symbol, self.track_style);
+        let thumb = (Some(self.thumb_symbol), self.thumb_style);
+        let end = (self.end_symbol, self.end_style);
+        let bar = iter::once(begin)
+            .chain(iter::repeat(track).take(track_start_len))
+            .chain(iter::repeat(thumb).take(thumb_len))
+            .chain(iter::repeat(track).take(track_end_len))
+            .chain(iter::once(end));
```

Comment:
Making begin and end `Option<(symbol, style)>` instead of making these (`Option<symbol>, style)` allows this to be 
```suggestion
        let bar = begin.into_iter()
            .chain(iter::repeat(track).take(track_start_len))
            .chain(iter::repeat(thumb).take(thumb_len))
            .chain(iter::repeat(track).take(track_end_len))
            .chain(end.into_iter());
```
This allows you to just iterate over the area, without having to keep an iterator variable manually (i) as the empty arrows are omitted from the iterator if they are `None`.